lib/FrameManager.js

```diff
@@ -326,8 +326,15 @@ class Frame {
    * @return {!Promise<!ElementHandle>}
    */
   async addScriptTag(options) {
-    if (typeof options.url === 'string')
-      return this._context.evaluateHandle(addScriptUrl, options.url);
+    if (typeof options.url === 'string') {
+      let handle;
+      try {
+        handle = await this._context.evaluateHandle(addScriptUrl, options.url);
```

Comment:
a few nits:
1. there should be no need to extract a "handle" variable
2. I'd extract the "url" into a separate variable since `options.url` might've changed by client code at the moment of throwing error.

```js
const url = options.url;
try {
  return await this._context.evaluateHandle(addScriptUrl, url);
} catch (error) {
  throw new Error(`Loading script from ${url} failed`);
}
```

Reply:
Thanks! This makes sense.